Classes/Controller/BackendController.php

```diff
@@ -224,7 +224,7 @@ protected function findNodeToEdit(): ?NodeInterface
         $reflectionMethod->setAccessible(true);
         $node = $reflectionMethod->invoke($this->backendRedirectionService, $siteNode->getContext()->getWorkspaceName());
 
-        if ($node === null) {
+        if ($node === null || !str_starts_with($node->findNodePath()->jsonSerialize(), $siteNode->findNodePath()->jsonSerialize())) {
```

Comment:
hmm we absolutely try to NOT to any string logic on the paths as this might be really fragile and with Neos 9 is no longer possible.
So really we want to fetch the node's side node and check if the site node equals `$siteNode`

Buuutt the proposed solution is slightly faster and doenst require any sql queries AND the last visited node feature was removed with Neos 9 meaning that this will not need to be 9.0 compatible.

Under these circumstances im willing to +1 this hack :D

Still one more thing, lets use the native methods:

```suggestion
        if ($node === null || !str_starts_with($node->getPath(), $siteNode->getPath())) {
```

Reply:
No objections to that :D
Just applied your proposal, thank you very much for the feedback :+1: